core/esmf-aspect-meta-model-java/src/main/java/org/eclipse/esmf/aspectmodel/loader/AspectModelLoader.java

```diff
@@ -480,6 +481,14 @@ public AspectModel loadAspectModelFiles( final Collection<AspectModelFile> input
             .filter( modelElement -> modelElement.is( Aspect.class ) )
             .findFirst()
             .ifPresent( aspect -> mergedModel.setNsPrefix( "", aspect.urn().getUrnPrefix() ) );
+      for ( AspectModelFile file : files ) {
+         if ( file.aspects().size() > 1 ) {
+            throw new InvalidCountOfAspectsException(
+                  "Invalid number of aspects for " + file
+                        + ". Total aspects: " + file.aspects().size()
+            );
+         }
+      }
```

Comment:
IMO this should throw an `AspectLoadingException`, not a special type of exception.

```suggestion
      for ( final AspectModelFile file : files ) {
         if ( file.aspects().size() > 1 ) {
            throw new AspectLoadingException(
                  "Aspect model file " + file.sourceLocation().map( location -> location + " ").orElse("") + "contains " + file.aspects().size() + " aspects, but may only contain one." );
         }
      }
```